### PR TITLE
HZN-1425: Generate events for every threshold violation

### PR DESCRIPTION
**Pull request by j-white**
JIRA: https://issues.opennms.org/browse/HZN-1425

### CODE REVIEW

File: opennms-doc/guide-admin/src/asciidoc/text/thresholds/configuring-thresholds.adoc [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+// Allow GitHub image rendering
+:imagesdir: ../../../images
+
+[[ga-threshd-configuration]]
+=== Configuring Thresholds
+
+Configuration of a threshold requires editing two main XML entities: <threshd-configuration/> and <thresholding-config/>.
+
+
+In Horizon 24.0.0, a two new attributes were added to the <threshd-configuration/> that directs the thresholders to continue to send events informing the current value of a triggered threshold.
```
Comment (mattixtech) [inline opennms-doc/guide-admin/src/asciidoc/text/thresholds/configuring-thresholds.adoc]:
```suggestion
In Horizon 24.0.0, two new attributes were added to the <threshd-configuration/> that directs the thresholders to continue to send events informing the current value of a triggered threshold.
```

File: opennms-doc/guide-admin/src/asciidoc/text/thresholds/configuring-thresholds.adoc [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+By default, these are not instantiated as Alarms.
+The two new new optional configuration attributes for this feature.
```
Comment (mattixtech) [inline opennms-doc/guide-admin/src/asciidoc/text/thresholds/configuring-thresholds.adoc]:
```suggestion
The two new optional configuration attributes for this feature.
```

File: opennms-doc/guide-admin/src/asciidoc/text/thresholds/introduction.adoc [hunk 3]
```diff
@@ -0,0 +1,13 @@
+
+// Allow GitHub image rendering
+:imagesdir: ../../../images
+
+[[ga-threshd-introduction]]
+_{opennms-product-name}_ has the ability to monitor the performance of a managed entity's resource(s) by using various thresholding techniques.
+These techniques, or algorithms, can be used applied against any metric data (telemetry) that has been gathered such as: CPU Load, Bandwidth, Disk space, etc. and, when triggered, causes Alarms to be generated.
```
Comment (mattixtech) [inline opennms-doc/guide-admin/src/asciidoc/text/thresholds/introduction.adoc]:
```suggestion
These techniques, or algorithms, can be applied against any metric data (telemetry) that has been gathered such as: CPU Load, Bandwidth, Disk space, etc. and, when triggered, causes Alarms to be generated.
```

File: opennms-doc/guide-admin/src/asciidoc/text/thresholds/threshd.adoc [hunk 4]
```diff
@@ -0,0 +1,7 @@
+
+// Allow GitHub image rendering
+:imagesdir: ../../../images
+
+=== Threshold Service Daemon
+Threshd, the Threshold Service Daemon, has the very complex task of keep state of performance of your corporate Key Performance Indicators (KPIs).
```
Comment (mattixtech) [inline opennms-doc/guide-admin/src/asciidoc/text/thresholds/threshd.adoc]:
```suggestion
Threshd, the Threshold Service Daemon, has the very complex task of tracking performance of your corporate Key Performance Indicators (KPIs).
```

File: opennms-services/src/main/java/org/opennms/netmgt/threshd/ThresholdEvaluatorHighLow.java [hunk 5]
```diff
@@ -236,8 +239,21 @@ public Event getEventForState(Status status, Date date, double dsValue, Collecti
                     return createBasicEvent(uei, date, dsValue, resource);
                 } else {
                     throw new IllegalArgumentException("Threshold type " + getThresholdConfig().getType() + " is not supported");
-                } 
-                
+                }
+
+            case SUSTAINED:
+                uei=getThresholdConfig().getSustainedUEI().orElse(null);
```
Comment (mattixtech) [inline opennms-services/src/main/java/org/opennms/netmgt/threshd/ThresholdEvaluatorHighLow.java]: Could run this block through auto-format

**Comment by j-white** [comment]
We've identified a different way of achieving the desired behavior that doesn't require these code changes. Putting on hold for now.

**Pull request closed without merging**